### DISCUSSION TITLE
Is there any way to pre-warmup the HNSW + MMAP searches?

### DISCUSSION THREAD

**Discussion by xiaobingxia-at** [milvus-io/milvus#38175] [Q&A and General discussion] [answered]
Hello, I'm using the HNSW + MMAP searches based on nvme local disk. However the searches usually are slow in the beginning, and the local disk read request per second is as high as 10k per second. But after continuously intensively searching for one hour, the local disk read request per second gradually goes down to zero, and the searches performance gets much faster, from 1 second to <15ms.

Just curious, is there any way to pre-warm up the HNSW + MMAP cache during data ingestion and data loading process?

**Comment by liliu-z** [comment]
1. Your pre-warm only cache the useful data since you are keeping searching with real cases (Only part of the index will get involved). But if Milvus do it during loading, it cannot tell which part need to be warm but try to load as much as possible to fill the memory
2. For 2.4.x we don't have a way to turn on this Milvus pre-ward, but 2.5.x can do it with a config change in Milvus' yml file.

**Reply by xiaobingxia-at** [reply]
what's the config in 2.5.x to turn on the pre-warm?

**Comment by xiaofan-luan** [comment, marked as answer]
> Hello, I'm using the HNSW + MMAP searches based on nvme local disk. However the searches usually are slow in the beginning, and the local disk read request per second is as high as 10k per second. But after continuously intensively searching for one hour, the local disk read request per second gradually goes down to zero, and the searches performance gets much faster, from 1 second to <15ms.
> 
> Just curious, is there any way to pre-warm up the HNSW + MMAP cache during data ingestion and data loading process?

trigger a search is the easiest way to do prewarm.

**Reply by xiaobingxia-at** [reply]
Per my understanding, trigger a search just pre-warm the partial index you need for this search only?
I usually need to do many millions of searches to bring down the query latency from 1 second to 10ms.
